src/routes/all-trips.js

```diff
@@ -8,20 +8,37 @@ export function get(req, res) {
 
   const beginners_only = req.query?.beginnersOnly === 'true'
   const showPrivate = res.locals.is_opo
+  const userId = req.user
 
-  const publicTrips = req.db.all(`
+  const trips = req.db.all(`
     SELECT trips.id, title, users.name as owner, location, start_time, end_time, description,
-      coalesce(clubs.name, 'None') as club
+      coalesce(clubs.name, 'None') as club,
+      trip_members.pending as status,
+      trip_members.leader as is_leader
     FROM trips
     LEFT JOIN users on trips.owner = users.id
     LEFT JOIN clubs on trips.club = clubs.id
+    LEFT JOIN trip_members on trips.id = trip_members.trip AND trip_members.user = ?
     WHERE start_time > ?
     ${showPrivate ? '' : 'AND private = 0'}
     ${beginners_only ? 'AND experience_needed = 0' : ''}
     ORDER BY start_time ASC
-  `, now.getTime())
+  `, userId, now.getTime()).map(trip => {
+    const formattedTrip = utils.formatForTripForTables(trip)
 
-  const trips = publicTrips.map(utils.formatForTripForTables)
+    // Set the status of the trip
+    if (trip.status === null) {
+      formattedTrip.status = 'not-joined'
+    } else if (trip.is_leader) {
+      formattedTrip.status = 'leader'
+    } else if (trip.status === 0) {
+      formattedTrip.status = 'approved'
+    } else {
+      formattedTrip.status = 'pending'
+    }
```

Comment:
I'd prefer to see this as a SQL case statement. I could also be convinced to add virtual column...